### PR TITLE
Create E2E tests for batch jobs

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Added tests for 

```
temporal workflow terminate --query ...
```

```
temporal workflow cancel --query ...
```

```
temporal workflow signal --query ...
```

## Why?

proper tests coverage for batch jobs. Needed to add now to address failing CIs here https://github.com/temporalio/cli/pull/201 

## Checklist

1. Closes 

2. How was this tested:

CI

3. Any docs updates needed?

### CODE REVIEW

File: tests/workflow_test.go [hunk 1]
```diff
@@ -114,5 +118,164 @@ func (s *e2eSuite) TestWorkflowUpdate() {
...
+			"non-existing-workflow-type",
+		)
+		s.NoError(err)
+	}
+
+	err := app.Run([]string{"", "workflow", "cancel", "--query", "WorkflowId = '1' OR WorkflowId = '2'", "--reason", "test", "--yes", "--namespace", testNamespace})
+	s.NoError(err)
+
+	// verify the job is complete
+	time.Sleep(1 * time.Second)
```
Comment (bergundy) [inline tests/workflow_test.go:148]: Use `s.Eventually` to make this less flaky.

File: tests/workflow_test.go [hunk 2]
```diff
@@ -114,5 +118,164 @@ func (s *e2eSuite) TestWorkflowUpdate() {
...
+	s.Equal(enums.BATCH_OPERATION_STATE_COMPLETED, deleteJob.State)
+
+	w1 := c.GetWorkflowHistory(context.Background(), "1", "", false, enums.HISTORY_EVENT_FILTER_TYPE_ALL_EVENT)
+	s.True(checkForEventType(w1, enums.EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED), "Workflow 1 should have received a signal")
+
+	w2 := c.GetWorkflowHistory(context.Background(), "2", "", false, enums.HISTORY_EVENT_FILTER_TYPE_ALL_EVENT)
+	s.True(checkForEventType(w2, enums.EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED), "Workflow 2 should have received a signal")
+
+	w3 := c.GetWorkflowHistory(context.Background(), "3", "", false, enums.HISTORY_EVENT_FILTER_TYPE_ALL_EVENT)
+	s.False(checkForEventType(w3, enums.EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED), "Workflow 3 should not have received a signal")
```
Comment (bergundy) [inline tests/workflow_test.go:216]: You can use a workflow that waits on a signal and check that it completed. It's a bit cleaner than inspecting the event history.

Comment (bergundy) [reply tests/workflow_test.go:216]: Same for cancelation above.

**Review by bergundy — APPROVED** [review]

File: tests/workflow_test.go (hunk 2 above)
Comment (feedmeapples) [reply tests/workflow_test.go:216]: right!

**Pull request merged**